spec/features/user_index_spec.rb

```diff
@@ -0,0 +1,28 @@
+require 'rails_helper'
+
+RSpec.describe 'User index page', type: :feature do
+  before(:each) do
+    @user = User.create(name: 'farouq', photo: 'image', bio: 'Developer from Egypt', email: 'test@email.com',
+                        password: 'password', confirmed_at: Time.now)
+    @post = Post.create(user: @user, title: 'My title', text: 'My text')
+    @like = Like.create(user: @user, post_id: @post.id)
+    @post.updates_posts_counter
+    visit new_user_session_path
+    fill_in 'Email', with: 'test@email.com'
+    fill_in 'Password', with: 'password'
+    click_button 'Log in'
+  end
+  describe 'tests for view users#index' do
+    it 'Can see the name of user' do
+      expect(page).to have_content 'farouq'
+    end
+
+    it 'I can see the profile picture for each user' do
+      expect(page.find('img')['src']).to have_content @user.photo
+    end
+
+    it 'number of posts of the user' do
+      expect(page).to have_content 'Number of posts: 1'
+    end
+  end
+end
```

Comment:
- [ ] Please add a test when the user clicks on the user redirect to the user's show page. [Requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Integration_specs_project.md#:~:text=When%20I%20click%20on%20a%20user%2C%20I%20am%20redirected%20to%20that%20user%27s%20show%20page.)